### ISSUE TITLE
cite with keyref to peer HTML broken in XHTML

### ISSUE THREAD

**Issue by robander** [dita-ot/dita-ot#1590]
Using the hierarchy.ditamap sample, I've added this key definition:

``` xml
<topicref 
   scope="peer" 
   format="html"
   keys="citekey" 
   href="../com.ibm.sample/sample.html#targetid" 
   navtitle="Peer HTML file"/>
```

In changingtheoil.xml, I've added these references:

``` xml
<p>Testing cite <cite keyref="citekey"/> and xref <xref keyref="citekey"/></p>
```

The xref element resolves correctly into a link to the HTML file, with the proper link text. The cite element generates `<a href="..%5C.html#targetid"><cite class="cite"/></a>` in the XHTML output, along with the following warning:

```
  [xslt] C:\DITA-OT1.8.M2\plugins\org.dita.xhtml\xsl\xslhtml\dita2htmlImpl.xsl:4243: Error! java.io.FileNotFoundException: C:\DITA-OT1.8.M2-\temp\temp20131023105911072\tasks\..\..\com.ibm.sample\sample.html (The system cannot find the path specified.) Cause: java.io.FileNotFoundException: C:\DITA-OT1.8.M2\temp\temp20131023105911072\tasks\..\..\com.ibm.sample\sample.html (The system cannot find the path specified.)
```

It looks like the keyref module does not do anything for cite, which makes some sense because cite does not have the scope/format/href attributes. It relies on the dita2html code, which seems to have some bad assumptions in the mode="find-keyref-target" -- it assumes the target is DITA, and assumes that a "." in the href is an extension to a local file.

**Comment by jelovirt** [comment]
@robander #1837 removed keyref processing from HTML. The output is not the same with the current code, but at least no errors are output.

**Comment by robander** [comment]
OK. "The output is not the same" - is it pulling link text now for cite? That was the primary issue - XSL error was almost secondary (though more obvious).

**Comment by robander** [comment]
Tried this again, just working with link text (no `@href` values). Tried a few things that might be expected to work. Storing text as:
* keyword in keywords: works (as it did when I opened this)
* navtitle element (works in Oxygen, not in DITA-OT)
* navtitle attribute (doesn't work)
* linktext element (doesn't work)
* in a `<cite>` (hidden inside phrase inside shortdesc) (doesn't work but I'm kind of glad it doesn't)

Test map:
```xml
<map xml:lang="en-us">
  <keydef keys="citkeyword">
  <topicmeta><keywords><keyword>text in keyword</keyword></keywords></topicmeta>
  </keydef>
  <keydef keys="citklinktext"><topicmeta><linktext>text in linktext</linktext></topicmeta></keydef>
  <keydef keys="citcit">
    <topicmeta>
      <shortdesc><ph><cite>WOW THIS IS UGLY, text in cit</cite></ph></shortdesc>
    </topicmeta>
  </keydef>
  <keydef keys="citnavatt" navtitle="navtitle attribute"/>
  <keydef keys="citnavel">
    <topicmeta><navtitle>navtitle element</navtitle></topicmeta>
  </keydef>
  <topicref href="cit-test.dita"/>
</map>
```

And the references in `cit-test.dita`:
```xml
<ul>
<li><cite keyref="citkeyword"/></li>
<li><cite keyref="citlinktext"/></li>
<li><cite keyref="citcit"/></li>
<li><cite keyref="citnavatt"/></li>
<li><cite keyref="citnavel"/></li>
</ul>
```

**Comment by robander** [comment]
Probably should have opened that as a new / general feature, not as a bug confused with the original peer HTML issue here (need to verify if that still exists).

**Comment by robander** [comment]
[Removed posts about other, broader issue.]

Original issue still exists here with 2.1, though the result differs. With the key definition in the original report, the HTML output now has the right link, the log has no error, but the link text doesn't come from `@navtitle` (it's just the relative path from `@href`).

**Comment by robander** [comment]
Fixed with #2719
